UnitTests/Dialogs/DialogTests.cs

```diff
@@ -1331,4 +1331,37 @@ public void Zero_Buttons_Works ()
 
         return (Begin (dlg), dlg);
     }
+
+    [Fact]
+    [AutoInitShutdown]
+    public void Run_Dispose_Dialog ()
+    {
+        var top = Top;
+
+        Dialog dlg = new ();
+
+        dlg.Ready += Dlg_Ready;
+
+        Run (dlg);
+
+#if DEBUG_IDISPOSABLE
+        Assert.True (dlg.WasDisposed);
+        Assert.True (Top.WasDisposed);
+        Assert.Equal (top, Top);
+#endif
+
+        // This is instance and the user is responsible to set to null or not
+        // because in the Application it's all working as expected.
+        // Fortunately, this instance is not null so that we can obtain the value of its properties
+        Assert.NotNull (dlg);
```

Comment:
Looking at this again... I'm confused.

This code literally says: "`dig` has been disposed, but it's ok to access it because the reference is not `null`".

This is not ok (and is actually the point I raised in filing #3338).

Reply:
That's the user instance and not the Top and the Current instance. If the user instance is null how you can get a property value after the Current is disposed?

Reply:
It's up to user disposing the instance and up to Application dispose the Current runstate.

Reply:
I think you are looking for an issue where there is none.